### ISSUE TITLE
[bug] Bazeldeps: build file is not correct for shared libary on Windows platform

### ISSUE THREAD

**Issue by kkpattern** [conan-io/conan#11347]
### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10
  * Compiler+version:
  * Conan version: 1.48.1
  * Python version: 3.10.1

### Steps to reproduce (Include if Applicable)
For shared libraries on Windows platform, the correct way to import them is using `interface_library` and `shared_library` together, like this:

```python
cc_import(
  name = "mylib",
  hdrs = ["mylib.h"],
  # mylib.lib is a import library for mylib.dll which will be passed to linker
  interface_library = "mylib.lib",
  # mylib.dll will be available for runtime
  shared_library = "mylib.dll",
)
```

Here is the [doc](https://docs.bazel.build/versions/main/be/c-cpp.html#cc_import).

Currently, bazel generator will generate BUILD file with shared libraries on Windows in this way:

```python
cc_import(
    name = "XXXprecompiled",
    shared_library = "lib/XXX.lib"
)
```

According to the conan [doc](https://docs.conan.io/en/latest/howtos/manage_shared_libraries/env_vars.html#building-an-application-using-the-shared-library-from-tool-a), DLL files are put in the bin directory. So we need to get corresponding DLL paths from bin directory.

Another problem we have is that DLL file of a library may depends on another DLL file. The publisher may put them all in the bin dir. But those DLL file won't be mention in the `libs` so bazel generator will skip those DLL libs.

We tried a super hacky way to solve this problem: We iterator the bin dir, if DLL file is encountered we list it as a shared library. If the corresponding interface lib is in `libs`, we will generate `cc_import` with both `interface_library` and `shared_library`. If only DLL file is present, we generate `cc_import` with `shared_library` only. Like this:

```python
cc_import(
    name = "XXXprecompiled",
    interface_library = "lib/XXX.lib",
    shared_library = "bin/XXX.dll"
)


cc_import(
    name = "YYYprecompiled",
    shared_library = "bin/YYY.dll"
)
```

This also solves another problem where the publisher may forget to the `shared` attribute to `True`. Bazel generator will assume the lib is static.

This solution actually works well internally in our project. But I feel it's too hacky for a general solution. If anyone is interested in our solution, I can create a branch and link it here.

**Comment by memsharded** [comment]
Hi @kkpattern 

Thanks for your feedback. When you say that you can share a branch, you mean with the purpose of merging it into the official release? Or you mean it is too hacky and this will not be possible.

To give some context, we were experiencing some related issues with the CMake generators, that were addressed in the 2.0 branch, because we were using there the ``package_type`` information to help reasoning about the type of libraries that a dependency contained, and that helped to create the targets. I guess something possible could be done here for ``BazelDeps``, though it should be targeted for 2.0 only at this moment.

**Comment by kkpattern** [comment]
> Hi @kkpattern
> 
> Thanks for your feedback. When you say that you can share a branch, you mean with the purpose of merging it into the official release? Or you mean it is too hacky and this will not be possible.

I mean it can be a implementation reference. The code is actually very simple and can be easily merge into official release. It's just too hacky that I'm not ensure whether it would work in other environment.

**Comment by memsharded** [comment]
Sure, if the code is simple and you know it, feel free to send a PR, and we will analyze it. Even if we cannot merge it, we will learn something, and hopefully we can at least consider it for the evolution of BazelDeps while moving to 2.0.

**Comment by kkpattern** [comment]
Cool! I will clean the code a bit and then create a pull request.

**Comment by kkpattern** [comment]
I created the pull request. I actually removed the hacky part to handle the libs with an incorrect `libs` setting or missing the `shared` option. For our internal use, we don't want to wait for other departments to upload a new corrected version. We want to get back to code as quickly as possible. So the hack is reasonable. But for official release, I think we want to make sure that the libs are published in the correct way?

**Comment by memsharded** [comment]
> We want to get back to code as quickly as possible. So the hack is reasonable. But for official release, I think we want to make sure that the libs are published in the correct way?

Not sure what you mean, the PR looks good enough, I am tentatively trying to include it in next 1.49. Do you think there is something else that needs to be done on our side before the release (note the BazelDeps is still experimental, it is not super-critical if the solution is not perfect, as long as it is an improvement, it is fine)

**Comment by kkpattern** [comment]
> Not sure what you mean,

I was talking about the hacky code I didn't commit. In my first commit in our internal conan repo, I was doing it differently. In that commit, I deliberately ignored the `libs` option, but iterate the `bin` directory directly. If a DLL file is found, I will check whether there is a corresponding lib. If so, I will use the `shared_library` plus `interface_library`. If not, I will use `shared_library` directly.

The reason I do this is because in our corp, some libs are uploaded "incorrectly". For example, some libraries will completely forget to set the `libs` option, so we need to deal with these libraries. But for official releases, I think it's up to the user who publish the library to make sure they have the `libs` and `shared` options set correctly. So I did not submit the corresponding hacky code in the pull request.

So I think we can merge the pull request. If we do want to handle the cases I mentioned above I can create another PR.

**Comment by memsharded** [comment]
PR merged, it will be in 1.49. Thanks!
